internal/config/env.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/gobuffalo/envy"
+)
+
+// ParseEnvironment loads a sibling `.env` file then looks through all environment
+// variables to set global configuration.
+func ParseEnvironment() error {
+	envy.Load()
+	azureEnv, _ := azure.EnvironmentFromName("AzurePublicCloud") // shouldn't fail
+
+	// these must be provided by environment
+	var err1, err2, err3, err4 error
+	clientID, err1 = envy.MustGet("AZURE_CLIENT_ID")
+	clientSecret, err2 = envy.MustGet("AZURE_CLIENT_SECRET")
+	tenantID, err3 = envy.MustGet("AZURE_TENANT_ID")
+	subscriptionID, err4 = envy.MustGet("AZURE_SUBSCRIPTION_ID")
+
+	for _, err := range []error{err1, err2, err3, err4} {
+		if err != nil {
+			return err
+		}
+	}
+
+	// we can choose defaults for these
+	groupName = envy.Get("AZURE_GROUP_NAME", "azure-go-samples")
```

Comment:
I'd make this random to reduce the chance that test runs are influenced by resources that may already exist.

``` Go
groupName = envy.Get("AZURE_GROUP_NAME", randname.GenerateWithPrefix("azure-go-samples-", 6)
```

Reply:
I was thinking of addressing the collision issue *and* the use of a global var for group name by deprecating `AZURE_GROUP_NAME` and using `AZURE_BASE_GROUP_NAME` instead, see next line. Then we can use `GenerateGroupName(affixes ...string)` in packages to get a derived name for local use.

I noted the deprecation in a comment on the accessor and will add a comment here too.